File: locals/solutions/parity/parityTimothy.c
#include<stdio.h>
#include<string.h>

int main(void) {
	char str[71];
	scanf("%s", str);
	long long parity = 0, targetEven = 0, targetOdd = 0;
	int len = strlen(str);
	for (int i = 0; i < len; i++) {
		long long bit = 1LL << (str[i] - 'a');
		parity ^= bit;
		targetOdd |= bit;
	}
	if (parity == targetEven) {
		printf("0\n");
	} else if (parity == targetOdd) {
		printf("1\n");
	} else {
		printf("2\n");
	}
	return 0;
}

File: locals/solutions/parity/parityTimothy.java
import java.util.*;

public class parityTimothy {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		String str = in.next();
		int parity = 0, targetEven = 0, targetOdd = 0;
		for (char c : str.toCharArray()) {
			int bit = 1 << (c - 'a');
			parity ^= bit;
			targetOdd |= bit;
		}
		if (parity == targetEven) {
			System.out.println(0);
		} else if (parity == targetOdd) {
			System.out.println(1);
		} else {
			System.out.println(2);
		}
	}
	
}